homeassistant/components/climate/__init__.py

```diff
@@ -465,11 +465,11 @@ async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set new target fan mode."""
         await self.hass.async_add_executor_job(self.set_fan_mode, fan_mode)
 
-    def set_hvac_mode(self, hvac_mode: str) -> None:
+    def set_hvac_mode(self, hvac_mode: HVACMode | str) -> None:
```

Comment:
The service schema will validate the parameter to the enum, right? Do we need the `str` type for compatibility if the method is called directly some other way?

Reply:
Not sure, that is why I kept it. Same as with the other typings. Until all occurrences in Core are replaced and resolved (after that, everything will still be backward compatible, even if we remove the legacy typing).

Reply:
Ok. 👍